lib/api/metadataSearch.js

```diff
@@ -0,0 +1,410 @@
+    <Name>example-bucket</Name>
+    <Prefix/>
+    <KeyCount>205</KeyCount>
+    <MaxKeys>1000</MaxKeys>
+    <IsTruncated>false</IsTruncated>
+    <Contents>
+        <Key>my-image.jpg</Key>
+        <LastModified>2009-10-12T17:50:30.000Z</LastModified>
+        <ETag>&quot;fba9dede5f27731c9771645a39863328&quot;</ETag>
+        <Size>434234</Size>
+        <StorageClass>STANDARD</StorageClass>
+    </Contents>
+</ListBucketResult>
+*/
+
+/*	Sample XML response for GET bucket objects:
+<ListBucketResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+  <Name>example-bucket</Name>
+  <Prefix></Prefix>
+  <Marker></Marker>
+  <MaxKeys>1000</MaxKeys>
+  <Delimiter>/</Delimiter>
+  <IsTruncated>false</IsTruncated>
+  <Contents>
+    <Key>sample.jpg</Key>
+    <LastModified>2011-02-26T01:56:20.000Z</LastModified>
+    <ETag>&quot;bf1d737a4d46a19f3bced6905cc8b902&quot;</ETag>
+    <Size>142863</Size>
+    <Owner>
+      <ID>canonical-user-id</ID>
+      <DisplayName>display-name</DisplayName>
+    </Owner>
+    <StorageClass>STANDARD</StorageClass>
+  </Contents>
+  <CommonPrefixes>
+    <Prefix>photos/</Prefix>
+  </CommonPrefixes>
+</ListBucketResult>
+*/
+
+/* eslint-disable max-len */
+// sample XML response for GET bucket object versions:
+// http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketGETVersion.html#RESTBucketGET_Examples
+/*
+<?xml version="1.0" encoding="UTF-8"?>
+<ListVersionsResult xmlns="http://s3.amazonaws.com/doc/2006-03-01">
+    <Name>bucket</Name>
+    <Prefix>my</Prefix>
+    <KeyMarker/>
+    <VersionIdMarker/>
+    <MaxKeys>5</MaxKeys>
+    <Delimiter>/</Delimiter>
+    <NextKeyMarker>>my-second-image.jpg</NextKeyMarker>
+    <NextVersionIdMarker>03jpff543dhffds434rfdsFDN943fdsFkdmqnh892</NextVersionIdMarker>
+    <IsTruncated>true</IsTruncated>
+    <Version>
+        <Key>my-image.jpg</Key>
+        <VersionId>3/L4kqtJl40Nr8X8gdRQBpUMLUo</VersionId>
+        <IsLatest>true</IsLatest>
+         <LastModified>2009-10-12T17:50:30.000Z</LastModified>
+        <ETag>&quot;fba9dede5f27731c9771645a39863328&quot;</ETag>
+        <Size>434234</Size>
+        <StorageClass>STANDARD</StorageClass>
+        <Owner>
+            <ID>75aa57f09aa0c8caeab4f8c24e99d10f8e7faeebf76c078efc7c6caea54ba06a</ID>
+            <DisplayName>mtd@amazon.com</DisplayName>
+        </Owner>
+    </Version>
+    <DeleteMarker>
+        <Key>my-second-image.jpg</Key>
+        <VersionId>03jpff543dhffds434rfdsFDN943fdsFkdmqnh892</VersionId>
+        <IsLatest>true</IsLatest>
+        <LastModified>2009-11-12T17:50:30.000Z</LastModified>
+        <Owner>
+            <ID>75aa57f09aa0c8caeab4f8c24e99d10f8e7faeebf76c078efc7c6caea54ba06a</ID>
+            <DisplayName>mtd@amazon.com</DisplayName>
+        </Owner>
+    </DeleteMarker>
+    <CommonPrefixes>
+        <Prefix>photos/</Prefix>
+    </CommonPrefixes>
+</ListVersionsResult>
+*/
+/* eslint-enable max-len */
+
+function processVersions(bucketName, listParams, list, encType) {
```

Comment:
The functions `processVersions`, `processMasterVersions`, `handleResult` are already in bucketGet.js, can we reuse them, so that we don't have to update 2 function when we update the GetBucket API?

Reply:
Also, because `skipUrlEncoding` is used in `processMasterVersions` only, no need to redefine it in this file.

Reply:
solved: https://github.com/scality/cloudserver/pull/4224/commits/f91cb730fb14781a5a3786de59750ffe0cb79251